src/OpenApi/test/Microsoft.AspNetCore.OpenApi.SourceGenerators.Tests/Helpers/HostFactoryResolver.cs

```diff
@@ -0,0 +1,363 @@
+
+    public const string BuildWebHost = nameof(BuildWebHost);
+    public const string CreateWebHostBuilder = nameof(CreateWebHostBuilder);
+    public const string CreateHostBuilder = nameof(CreateHostBuilder);
+    private const string TimeoutEnvironmentKey = "DOTNET_HOST_FACTORY_RESOLVER_DEFAULT_TIMEOUT_IN_SECONDS";
+
+    // The amount of time we wait for the diagnostic source events to fire
+    private static readonly TimeSpan s_defaultWaitTimeout = SetupDefaultTimeout();
+
+    private static TimeSpan SetupDefaultTimeout()
+    {
+        if (Debugger.IsAttached)
+        {
+            return Timeout.InfiniteTimeSpan;
+        }
+
+        if (uint.TryParse(Environment.GetEnvironmentVariable(TimeoutEnvironmentKey), out uint timeoutInSeconds))
+        {
+            return TimeSpan.FromSeconds((int)timeoutInSeconds);
+        }
+
+        return TimeSpan.FromMinutes(5);
+    }
+
+    public static Func<string[], TWebHost> ResolveWebHostFactory<TWebHost>(Assembly assembly)
+    {
+        return ResolveFactory<TWebHost>(assembly, BuildWebHost);
+    }
+
+    public static Func<string[], TWebHostBuilder> ResolveWebHostBuilderFactory<TWebHostBuilder>(Assembly assembly)
+    {
+        return ResolveFactory<TWebHostBuilder>(assembly, CreateWebHostBuilder);
+    }
+
+    public static Func<string[], THostBuilder> ResolveHostBuilderFactory<THostBuilder>(Assembly assembly)
+    {
+        return ResolveFactory<THostBuilder>(assembly, CreateHostBuilder);
+    }
+
+    // This helpers encapsulates all of the complex logic required to:
+    // 1. Execute the entry point of the specified assembly in a different thread.
+    // 2. Wait for the diagnostic source events to fire
+    // 3. Give the caller a chance to execute logic to mutate the IHostBuilder
+    // 4. Resolve the instance of the applications's IHost
+    // 5. Allow the caller to determine if the entry point has completed
+    public static Func<string[], object> ResolveHostFactory(Assembly assembly,
+                                                             TimeSpan waitTimeout = default,
+                                                             bool stopApplication = true,
+                                                             Action<object> configureHostBuilder = null,
+                                                             Action<Exception> entrypointCompleted = null)
+    {
+        if (assembly.EntryPoint is null)
+        {
+            return null;
+        }
+
+        try
+        {
+            // Attempt to load hosting and check the version to make sure the events
+            // even have a chance of firing (they were added in .NET >= 6)
+            var hostingAssembly = Assembly.Load("Microsoft.Extensions.Hosting");
+            if (hostingAssembly.GetName().Version is Version version && version.Major < 6)
```

Comment:
still relevant?

Reply:
Yep -- we can get rid of this. This source is lifted [from here](https://github.com/dotnet/runtime/blob/93dd79997031a32e4034825bd671990dbc1e4ceb/src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs) so this handle isn't relevant for our particular test setup.